app/controllers/gemstones_controller.rb

```diff
@@ -0,0 +1,35 @@
+class GemstonesController < ApplicationController
+
+  def index
+    @gemstones = Gemstone.all
+    respond_to do |format|
+      format.html
+      format.json { render json: {url: gemstones_path, gemstones: @gemstones } }
+    end
+  end
+
+  def show
+    @gemstone = Gemstone.find(params[:id])
+    respond_to do |format|
+      format.html
+      format.json { render json: {url: gemstone_path, gemstone: @gemstone } }
```

Comment:
This works, and is a very readable format of your json response. I'd say if a 3rd key is added, then it's time to switch to jbuilder or ActiveModelSerializer.